docs/topics/gradle/gradle-configure-project.md

```diff
@@ -123,19 +128,66 @@ In the build module, you may have related compile tasks, for example:
 >
 {type="note"}
 
-For related tasks like these, the Kotlin Gradle plugin checks for JVM target compatibility. Different values of `jvmTarget` in
-the `kotlin` extension and [`targetCompatibility`](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)
-in the `java` extension cause JVM target incompatibility. For example:
+For related tasks like these, the Kotlin Gradle plugin checks for JVM target compatibility. Different values of 
+the [`jvmTarget` attribute](gradle-compiler-options.md#attributes-specific-to-jvm) in the `kotlin` extension or task 
+and [`targetCompatibility`](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)
+in the `java` extension or task cause JVM target incompatibility. For example:
 the `compileKotlin` task has `jvmTarget=1.8`, and
 the `compileJava` task has (or [inherits](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)) `targetCompatibility=15`.
 
 Configure the behavior of this check by setting the `kotlin.jvm.target.validation.mode` property in the `build.gradle`
 file to:
 
-* `warning` – the default value; the Kotlin Gradle plugin will print a warning message.
-* `error` – the plugin will fail the build.
+* `error` – the plugin will fail the build; the default value for projects on Gradle 8.0+.
+* `warning` – the Kotlin Gradle plugin will print a warning message; the default value for projects on Gradle less than 8.0.
 * `ignore` – the plugin will skip the check and won't produce any messages.
 
+To avoid JVM target incompatibility, [configure a toolchain](#gradle-java-toolchains-support) or align JVM versions manually.
+
+#### What can go wrong if not checking targets compatibility {initial-collapse-state="collapsed"}
+
+There are two ways of manually setting JVM targets for Kotlin and Java source sets:
+* The implicit way via [setting up a Java toolchain](#gradle-java-toolchains-support).
+* The explicit way via setting the `jvmTarget` attribute in the `kotlin` extension or task and `targetCompatibility` 
+  in the `java` extension or task.
+
+JVM target incompatibility occur if you:
```

Comment:
@Tapchicoma 
You said (this comment disappeared because I made a change) this part was hard to read and proposed deleting it and starting with the example. Well, I want to offer another variant: I tried to make the text more readable, [here](https://github.com/JetBrains/kotlin-web-site/pull/3333/commits/f90e40c026c6a5664a70432b4a07f9b1140b0088) you can see the change. I wouldn't like to start with an example, I always try to write some introductory sentences. Hope it works now; let me know if it's still hard to follow.

Reply:
yeah - now looks better. Let's leave it as it is now